### PR TITLE
MAISTRA-1898: Update to latest upstream 1.6

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 1]
```diff
@@ -301,6 +309,7 @@ func NewController(client kubernetes.Interface, metadataClient metadata.Interfac
 
 	c.services = cache.NewSharedIndexInformer(svcMlw, &v1.Service{}, options.ResyncPeriod,
 		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+	c.serviceLister = listerv1.NewServiceLister(c.services.GetIndexer())
```
Comment (jwendell) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:312]: @dgn @bison ^

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 2]
```diff
@@ -330,7 +339,16 @@ func NewController(client kubernetes.Interface, metadataClient metadata.Interfac
 		registerHandlers(c.filteredNodeInformer, c.queue, "Nodes", c.onNodeEvent)
 	}
 
-	c.pods = newPodCache(c, mrc, options)
+	c.pods = newPodCache(c, mrc, options, func(key string) {
```
Comment (jwendell) [inline pilot/pkg/serviceregistry/kube/controller/controller.go:342]: @dgn @bison

**Comment by jwendell** [comment]
@dgn @bison I've tagged you both in the non-trivial conflicts I had to fix. Please take a look and let me know if it's wrong.

File: pilot/pkg/serviceregistry/kube/controller/controller.go (hunk 1 above)
Comment (bison) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:312]: If this is just in reference to the lister, it looks fine. As long as the lister is based on an informer using the MLW, it should work like we expect it to.

File: pilot/pkg/serviceregistry/kube/controller/controller.go (hunk 2 above)
Comment (bison) [reply pilot/pkg/serviceregistry/kube/controller/controller.go:342]: Looks fine to me.

**Comment by dgn** [comment]
I'm working on fixing the tests

**Comment by jwendell** [comment]
@dgn ok, I was about to start this task. Thanks.

**Comment by dgn** [comment]
Okay, I pushed a commit that should fix the unit test failures. The integration test failure looks like a flake to me (it's a test that relies on `sleep()` to validate cert lifetimes...)

**Review by dgn — APPROVED** [review]

**Review by bison — APPROVED** [review]
Obviously mostly skimmed this... but if the tests pass, then LGTM.

**Pull request merged**